modules/neuroimaging/denoising/nlmeans/main.nf

```diff
@@ -0,0 +1,51 @@
+
+process DENOISING_NLMEANS {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://scil.usherbrooke.ca/containers/scilus_2.0.0.sif':
+        'scilus/scilus:2.0.0' }"
+
+    input:
+    tuple val(meta), path(image), path(mask)
+
+    output:
+    tuple val(meta), path("*_denoised.nii.gz")      , emit: image
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def args = []
+    if (mask) args += ["--mask $mask"]
+
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+
+    scil_denoising_nlmeans.py $image ${prefix}_denoised.nii.gz 1 ${args.join(" ")}
```

Comment:
```suggestion
    scil_denoising_nlmeans.py \\
        $image \\
        ${prefix}.nii.gz \\
        1 \\
        $mask_arg \\
        $args
```
This makes the command a bit more readable. Can you also add some comments alongside the `1` to explain what it's purpose is? I've also removed the custom suffix from the output file to adhere to the nf-core guidelines. You can easily re-add this suffix in your pipeline configuration file using this:
```
process {
    withName: DENOISING_NLMEANS {
        ext.prefix = { "${meta.id}_denoised" }
    }
}
```